tests/functional/TestSmokeTest/tests/block_volume_test.go

```diff
@@ -133,3 +136,120 @@ func TestBlockVolumeCreateManyAtOnce(t *testing.T) {
 			"expected r.err == nil, got:", r.err, "@", r.index)
 	}
 }
+
+func TestBlockVolumeAlreadyDeleted(t *testing.T) {
+
+	// Setup the VM storage topology
+	setupCluster(t, 3, 4)
+	defer teardownCluster(t)
+	defer teardownBlock(t)
+
+	blockReq := &api.BlockVolumeCreateRequest{}
+	blockReq.Size = 3
+	blockReq.Hacount = 3
+
+	type result struct {
+		index int
+		err   error
+	}
+
+	bvol, err := heketi.BlockVolumeCreate(blockReq)
+	tests.Assert(t, err == nil, "expected err == nil, got:", err)
+
+	hvol, err := heketi.VolumeInfo(bvol.BlockHostingVolume)
+	tests.Assert(t, err == nil, "expected err == nil, got:", err)
+
+	host := storage0ssh
+	cmd := fmt.Sprintf("gluster-block delete %v/%v --json",
+		hvol.Name, bvol.Name)
+	s := ssh.NewSshExecWithKeyFile(
+		logger, "vagrant", "../config/insecure_private_key")
+	o, err := s.ConnectAndExec(host, []string{cmd}, 10, true)
+	tests.Assert(t, err == nil, "expected err == nil, got:", err)
+	tests.Assert(t, len(o) == 1, "expected len(o) == 1, got:", len(o))
+
+	err = heketi.BlockVolumeDelete(bvol.Id)
+	tests.Assert(t, err == nil, "expected err == nil, got:", err)
+}
+
+func TestBlockVolumeDeleteFailureConditions(t *testing.T) {
+
+	// Setup the VM storage topology
+	setupCluster(t, 3, 4)
+	defer teardownCluster(t)
+	defer teardownBlock(t)
+
+	blockReq := &api.BlockVolumeCreateRequest{}
+	blockReq.Size = 3
+	blockReq.Hacount = 3
+
+	type result struct {
+		index int
+		err   error
+	}
+
+	bvol, err := heketi.BlockVolumeCreate(blockReq)
+	tests.Assert(t, err == nil, "expected err == nil, got:", err)
+
+	_, err = heketi.BlockVolumeInfo(bvol.Id)
+	tests.Assert(t, err == nil, "expected err == nil, got:", err)
+
+	//targetHost := storage0ssh
+	downHosts := []string{storage1ssh, storage2ssh}
+
+	var cmds []string
+	s := ssh.NewSshExecWithKeyFile(
+		logger, "vagrant", "../config/insecure_private_key")
+	stopServices := func(both bool) {
+		logger.Info("Stopping services for test")
+		for _, host := range downHosts {
+			cmds = []string{
+				"systemctl stop gluster-blockd",
+			}
+			if both {
+				cmds = append(cmds, "systemctl stop glusterd")
+			}
+			_, err := s.ConnectAndExec(host, cmds, 10, true)
+			tests.Assert(t, err == nil, "expected err == nil, got:", err)
+		}
+		time.Sleep(time.Second * 5)
+	}
+	startServices := func() {
+		logger.Info("Starting services back up")
+		for _, host := range downHosts {
+			cmds = []string{
+				"systemctl start glusterd",
+				"systemctl start gluster-blockd",
+			}
+			_, err := s.ConnectAndExec(host, cmds, 10, true)
+			tests.Assert(t, err == nil, "expected err == nil, got:", err)
+		}
+		time.Sleep(time.Second * 5)
+	}
+
+	defer startServices()
+
+	l, err := heketi.BlockVolumeList()
+	tests.Assert(t, err == nil, "expected err == nil, got:", err)
+	tests.Assert(t, len(l.BlockVolumes) == 1,
+		"expected len(l.BlockVolumes) == 1, got:", len(l.BlockVolumes))
+
+	stopServices(false)
+	for i := 0; i < 10; i++ {
```

Comment:
Could you add a comment explaining the reason for doing a loop for 10 attempts here?